<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>Introduction</title>
  <style>
    html {
      line-height: 1.5;
      font-family: Georgia, serif;
      font-size: 12pt;
      color: #1a1a1a;
      background-color: #fdfdfd;
    }
    body {
      margin: 0 auto;
      max-width: 36em;
      padding-left: 50px;
      padding-right: 50px;
      padding-top: 50px;
      padding-bottom: 50px;
      hyphens: auto;
      overflow-wrap: break-word;
      text-rendering: optimizeLegibility;
      font-kerning: normal;
    }
    @media (max-width: 600px) {
      body {
        font-size: 0.9em;
        padding: 1em;
      }
      h1 {
        font-size: 1.8em;
      }
    }
    @media print {
      body {
        background-color: transparent;
        color: black;
        font-size: 12pt;
      }
      p, h2, h3 {
        orphans: 3;
        widows: 3;
      }
      h2, h3, h4 {
        page-break-after: avoid;
      }
    }
    p {
      margin: 1em 0;
    }
    a {
      color: #1a1a1a;
    }
    a:visited {
      color: #1a1a1a;
    }
    img {
      max-width: 100%;
    }
    h1, h2, h3, h4, h5, h6 {
      margin-top: 1.4em;
    }
    h5, h6 {
      font-size: 1em;
      font-style: italic;
    }
    h6 {
      font-weight: normal;
    }
    ol, ul {
      padding-left: 1.7em;
      margin-top: 1em;
    }
    li > ol, li > ul {
      margin-top: 0;
    }
    blockquote {
      margin: 1em 0 1em 1.7em;
      padding-left: 1em;
      border-left: 2px solid #e6e6e6;
      color: #606060;
    }
    code {
      font-family: Menlo, Monaco, 'Lucida Console', Consolas, monospace;
      font-size: 85%;
      margin: 0;
    }
    pre {
      margin: 1em 0;
      overflow: auto;
    }
    pre code {
      padding: 0;
      overflow: visible;
      overflow-wrap: normal;
    }
    .sourceCode {
     background-color: transparent;
     overflow: visible;
    }
    hr {
      background-color: #1a1a1a;
      border: none;
      height: 1px;
      margin: 1em 0;
    }
    table {
      margin: 1em 0;
      border-collapse: collapse;
      width: 100%;
      overflow-x: auto;
      display: block;
      font-variant-numeric: lining-nums tabular-nums;
    }
    table caption {
      margin-bottom: 0.75em;
    }
    tbody {
      margin-top: 0.5em;
      border-top: 1px solid #1a1a1a;
      border-bottom: 1px solid #1a1a1a;
    }
    th {
      border-top: 1px solid #1a1a1a;
      padding: 0.25em 0.5em 0.25em 0.5em;
    }
    td {
      padding: 0.125em 0.5em 0.25em 0.5em;
    }
    header {
      margin-bottom: 4em;
      text-align: center;
    }
    #TOC li {
      list-style: none;
    }
    #TOC ul {
      padding-left: 1.3em;
    }
    #TOC > ul {
      padding-left: 0;
    }
    #TOC a:not(:hover) {
      text-decoration: none;
    }
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Introduction</h1>
</header>
<nav id="TOC" role="doc-toc">
<ul>
<li><a href="#prognostic-biomarkers"
id="toc-prognostic-biomarkers">Prognostic biomarkers</a>
<ul>
<li><a href="#biomarkers-based-on-histopathology"
id="toc-biomarkers-based-on-histopathology">Biomarkers based on
histopathology</a></li>
<li><a href="#computation-pathology"
id="toc-computation-pathology">Computation Pathology</a></li>
</ul></li>
<li><a href="#convolutional-neural-networks"
id="toc-convolutional-neural-networks">Convolutional neural
networks</a></li>
<li><a href="#weakly-supervised-methods"
id="toc-weakly-supervised-methods">Weakly supervised methods</a></li>
<li><a href="#thesis-overview" id="toc-thesis-overview">Thesis
overview</a></li>
</ul>
</nav>
<p>This thesis will focus on the most common form of prostate cancer,
prostatic adenocarcinoma. For simplicity, we will refer to this type as
just ‘prostate cancer.’ This disease affects one in .. men every year,
making it the most prevalent common type of cancer in men (excluding
skin cancers). Prostate cancer is a disease of the epithelial cells of
the prostate. Epithelial cells line our body cavities, our hollow
organs, and glands. They undergo rapid proliferation, primarily due to
damage. This proliferation increases the risk of genetic mutations,
ultimately increasing the risk of a cell uncontrollably dividing.
Together with enabling factors of its tissue environment, this can give
rise to cancer.</p>
<p>In general, the more aggressive cancerous cells are, the less they
will behave and morphologically appear like their original function. The
prostate is a gland that produces prostatic fluid. The fluid is
transported to the urethra by small 0ubes. These tubes, called prostatic
glands, are lined with epithelium. Low-grade cancer will thus mimic
those gland structures. High-grade prostate cancer loses its structural
morphology, forming sheets of cells or even quasi-randomly dispersed
individual cancerous cells.</p>
<p>American pathologist Donald Floyd Gleason systematically wrote down
the correlation between growth patterns and prognosis in prostate cancer
in the 1960s<span class="citation"
data-cites="cite_original"><sup><strong>cite_original?</strong></sup></span>.
Pathologists still use this Gleason grading, albeit several revisions
later<span class="citation"
data-cites="Epstein2016-im"><sup>1</sup></span>, to classify prostate
cancer.</p>
<p>[Insert image Gleason grown patterns]</p>
<h1 id="prognostic-biomarkers">Prognostic biomarkers</h1>
<p>To decide on a treatment plan, clinicians divide patients into risk
groups according to traditional baseline characteristics, such as PSA
blood level, Gleason grade, tumor location, tumor size, and lymph node
status.[lam2019] This information is gathered from histopathological,
radiological assessment, and lab assessments. These assessments can be
considered biomarkers as they indicate the prognosis of a patient<span
class="citation" data-cites="chen2011"><sup>2</sup></span>. The more
precise these assessments are, the better we can tailor the treatment to
the specific patient; this is known as personalized medicine.</p>
<p>To make treatment more tailored to the patient, researchers try to
develop new biomarkers. There is a demand for new biomarkers because
most prostate cancers progress so slowly that they are unlikely to
threaten the affected individual’s survival, and patients with the same
histological and clinical characteristics, can have strikingly different
outcomes<span class="citation"
data-cites="cucchiara2018"><sup>3</sup></span>. Being able to pick out
patients with good prognostis would improve their quality of life since
treatments for prostate cancer obviously have adverse effects
(#tab:adverse){reference-type=“ref” reference=“tab:adverse”}). Equally
so for patients for which we can find out the treatment will not
contribute to their health. To prevent adverse effects and increase
treatment response, researchers are developing new markers in
genomics<span class="citation"
data-cites="cucchiara2018"><sup>3</sup></span>, radiology<span
class="citation" data-cites="roest2023"><sup>4</sup></span>, and
pathology, the latter of which is the subject of this thesis.</p>
<div id="tab:adverse">
<table>
<caption>Common Prostate Cancer Treatment Options and Potential Adverse
Effects, reproduced from Dunn et al.<span class="citation"
data-cites="dunn2011"><sup>5</sup></span></caption>
<tbody>
<tr class="odd">
<td style="text-align: left;"><strong>Treatment Option</strong></td>
<td style="text-align: left;"><strong>Disease Progression</strong></td>
<td style="text-align: left;"><strong>Potential Adverse
Effects</strong></td>
</tr>
<tr class="even">
<td style="text-align: left;">Active surveillance</td>
<td style="text-align: left;">Localized</td>
<td style="text-align: left;">Illness uncertainty</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Radical prostatectomy</td>
<td style="text-align: left;">Localized</td>
<td style="text-align: left;">Erectile dysfunction</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Urinary incontinence</td>
</tr>
<tr class="odd">
<td style="text-align: left;">External beam radiation</td>
<td style="text-align: left;">Localized and advanced disease</td>
<td style="text-align: left;">Urinary urgency and frequency, dysuria,
diarrhea and proctitis</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Erectile dysfunction</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Urinary incontinence</td>
</tr>
<tr class="even">
<td style="text-align: left;">Brachytherapy</td>
<td style="text-align: left;">Localized</td>
<td style="text-align: left;">Urinary urgency and frequency, dysuria,
diarrhea and proctitis</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Erectile dysfunction</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Urinary incontinence</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Cryotherapy</td>
<td style="text-align: left;">Localized</td>
<td style="text-align: left;">Erectile dysfunction</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Urinary incontinence and retention</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Rectal pain and fistula</td>
</tr>
<tr class="even">
<td style="text-align: left;">Hormone therapy</td>
<td style="text-align: left;">Advanced</td>
<td style="text-align: left;">Fatigue</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Hot flashes, and flare effect</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Hyperlipidemia</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Insulin resistance</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Cardiovascular disease</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Anemia</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Osteoporosis</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Erectile dysfunction</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Cognitive deficits</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Chemotherapy</td>
<td style="text-align: left;">Advanced</td>
<td style="text-align: left;">Myelosuppression</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Hypersensitivity reaction</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Gastrointestinal upset</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Peripheral neuropathy</td>
</tr>
</tbody>
</table>
</div>
<h2 id="biomarkers-based-on-histopathology">Biomarkers based on
histopathology</h2>
<p>We know that histopathology holds prognostic information. Commonly,
pathologist also report extra-capsular extension of the tumor and
perineural invasion, both signs of poor prognosis. As mentioned earlier,
the Gleason patterns were discovered by recording patient prognosis.
Gleason growth patterns are grouped into five different groups, of which
current pathologists mainly use the last three. It’s not hard to imagine
there being more clues in the morphology of the behavior of the tumor,
if only because the landscape of prostate cancer growth patterns is
certainly more complex than the three groups we divide them in. Of note,
recently, the ‘subpattern’ cribriform-like growth was discovered to be
an aggressive pattern.</p>
<p>However, these visual biomarkers are hard to explicitly specify and
quantify manually. Luckily, machine learning can help. The first chapter
will discuss this approach further. However, it makes sense to introduce
this research field, computational pathology, first.</p>
<h2 id="computation-pathology">Computation Pathology</h2>
<p>Pathology is undergoing a digital revolution. More and more labs are
purchasing whole-slide scanners, with some already reading most slides
digitally. Glass slides are digitized, resulting in gigapixel digital
images, commonly referred to as whole-slide images (WSIs). Once the data
is digital, opportunities for computational analysis and assistance
arise.</p>
<p>Litjens et al. [1] gave an overview of deep learning applications in
computational pathology up to 2016. Some early successes in the field
focused on segmentation, tissue classification, and disease
classification. Not on prognosis or treatment response prediction
treatment. Likely due to the fact these tasks are relatively easier and
since this kind of data is relatively cheap to obtain compared to
prognosis information</p>
<!--
More recent work focuses on clinical implementation and validation. Campanella
et al. [2] propose a system for prostate cancer detection trained with weak
slide-level labels. Bulten et al. [3] validate a system to determine the
Gleason grade of prostate cancer biopsies, showing human-level performance.   

Especially interesting for clinical implementation is the ability to learn with
weaker labels, circumventing expensive precise annotations. Although labels
directly from pathology reports are noisy, the information is good enough to
guide treatment decisions. As such, recent focus, like in this thesis, is to
train deep learning models directly using these readily available labels.

Overall, deep learning shows potential to enhance and assist pathology using
the digital pathology slides. The following chapters will dive deeper into the
specific computational pathology tasks tackled in this thesis.
-->
<h1 id="convolutional-neural-networks">Convolutional neural
networks</h1>
<p>Convolutional neural networks (CNNs) have emerged among the
state-of-the-art machine learning algorithms for various computer vision
tasks, such as image classification and segmentation.</p>
<p>The central component of a convolutional neural network is often
represented as a sliding kernel (or filter) over an input matrix,
producing an output matrix. This has several advantages; we can use a
smaller kernel than the whole making the network less complex while
exploiting the fact that objects in the image are translation invariant.
A cat in the upper-left corner is still a cat in the lower right corner.
We introduce this inductive bias to the network by using
convolutions.</p>
<p>Most convolutional neural network architectures have alternating
blocks of layers consisting of a convolutional operation, a non-linear
activation function, and often a normalization operation. The
non-linearities are essential, as they make the networks able to
represent more complex (non-linear) functions. Normalization layers
bound the output of the block to be within a specific range which helps
during the optimization of the network.</p>
<p>Even though sliding kernels are less complex than having one
parameter per input value, the network architectures have evolved to
become deeper and wider to enhance their accuracy further. Training
larger CNNs demands larger amounts of computer memory, which increases
exponentially with the size of input images. Consequently, most natural
image datasets in computer vision, such as ImageNet and CIFAR-10,
contain sub-megapixel images to circumvent memory limitations.</p>
<p>In specific domains like remote sensing and medical imaging, there is
a need to train CNNs on high-resolution, where most of the information
is contained. Ideally we want to combine the high-resolution information
with more global context, as pathologists can do during daily practive.
However, computer memory becomes a limiting factor. The memory
requirements of CNNs increase proportionally to the input size of the
network, quickly filling up memory with multi-megapixel images. As a
result, only small CNNs can be trained with such images, rendering
state-of-the-art architectures unattainable even on large computing
clusters.</p>
<h1 id="weakly-supervised-methods">Weakly supervised methods</h1>
<p>For others, several authors have suggested approaches to train
convolutional neural networks (CNNs) with large input images while
preventing memory bottlenecks. Their methods can be roughly grouped into
three categories: (A) altering the dataset, (B) altering usage of the
dataset, and (C) altering the network or underlying implementations.</p>
<pre><code>As a result of the memory limitations, most networks are trained on small
patches from the WSI. This patch-based training requires detailed pixel-level
annotations of the classes, such as outlines of tumor regions by an expert
pathologist. However, pixel-level labeling is expensive, time-consuming, and
not routinely performed in clinical practice.

To overcome this, researchers have developed weakly supervised methods that can
learn from readily available slide-level labels in pathology reports, without
expensive patch-level annotations. A popular approach is multiple instance
learning (MIL). In MIL, a CNN is trained on patches under the assumption that a
positive slide contains at least one positive patch. Only the most informative
patch per slide is used for backpropagation. However, this limits the
field-of-view and contextual information.

Another weakly supervised approach is to compress the WSI into a
lower-dimensional latent space using autoencoders. The embedding can be used to
train a model per patch. However, useful information may be lost during
compression. Also, relationships between patches are disregarded.

Methods like recurrent attention networks try to increase the field-of-view by
analyzing multiple patches. However, memory constraints limit the patch size,
and patches must be kept in memory to calculate attention weights. Overall,
most current weakly supervised methods rely on patches, constraining the
network&#39;s field-of-view.

In this thesis, a novel streaming method is proposed to train CNNs end-to-end
on entire WSIs with slide-level labels. By reconstructing activations
tile-by-tile, streaming removes the need to crop images. A CNN can learn from
full contextual information at high resolution, without relying on patches.
Experiments show streaming reaches performance on par with patch-based methods
needing more supervision. Thus, streaming enables direct learning from
morphology to aid histopathology analysis using readily available slide-level
labels. Chapter 2 will go more into depth on this.</code></pre>
<h1 id="thesis-overview">Thesis overview</h1>
<pre><code>Chapter 2 proposes a method called &quot;streaming&quot; to train convolutional neural
networks end-to-end on multi-megapixel histopathology images, circumventing
memory limitations. We tiles the input image and reconstructs activations,
allowing the use of entire high-resolution images during training without
cropping. 

Chapter 3 applies streaming to train models on whole prostate biopsy images
using only slide-level labels from pathology reports. It shows a modern CNN can
learn from high-resolution images without patch-level annotations. The method
reaches similar performance to state-of-the-art patch-based and multiple
instance learning techniques.

Chapter 4 demonstrates a deep learning system to predict biochemical recurrence
of prostate cancer using tissue morphology. Trained on a nested case-control
study and validated on an independent cohort, the system finds patterns
predictive of recurrence beyond standard Gleason grading. Concept-based
explanations show tissue features aligned with pathologist interpretation.

In summary, the thesis explores computational pathology methods to analyze
entire high-resolution histopathology images despite memory constraints. It
shows neural networks can learn from morphology to aid prostate cancer
diagnosis and prognosis when trained end-to-end on whole images using readily
available slide-level labels.</code></pre>
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-Epstein2016-im" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div
class="csl-right-inline">Epstein JI, Egevad L, Amin MB, Delahunt B,
Srigley JR, Humphrey PA. The 2014 international society of urological
pathology (<span>ISUP</span>) consensus conference on gleason grading of
prostatic carcinoma. 2016;40:244-252.</div>
</div>
<div id="ref-chen2011" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Chen
XH, Huang S, Kerr D. Biomarkers in clinical medicine. Published online
2011.</div>
</div>
<div id="ref-cucchiara2018" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div
class="csl-right-inline">Cucchiara V, Cooperberg MR, Dall’Era M, et al.
Genomic <span>Markers</span> in <span>Prostate Cancer Decision
Making</span>. <em>European Urology</em>. 2018;73(4):572-582. doi:<a
href="https://doi.org/10.1016/j.eururo.2017.10.036">10.1016/j.eururo.2017.10.036</a></div>
</div>
<div id="ref-roest2023" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div
class="csl-right-inline">Roest C, Kwee TC, Saha A, Fütterer JJ, Yakar D,
Huisman H. <span class="nocase">AI-assisted</span> biparametric
<span>MRI</span> surveillance of prostate cancer: Feasibility study.
<em>European Radiology</em>. 2023;33(1):89-96. doi:<a
href="https://doi.org/10.1007/s00330-022-09032-7">10.1007/s00330-022-09032-7</a></div>
</div>
<div id="ref-dunn2011" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Dunn
MW, Kazer MW. Prostate <span>Cancer Overview</span>. <em>Seminars in
Oncology Nursing</em>. 2011;27(4):241-250. doi:<a
href="https://doi.org/10.1016/j.soncn.2011.07.002">10.1016/j.soncn.2011.07.002</a></div>
</div>
</div>
</body>
</html>
